test/test_backend.py

```diff
@@ -0,0 +1,20 @@
+import warnings
+from unittest.mock import MagicMock
+
+from joblibspark.backend import SparkDistributedBackend
+
+
+def test_effective_n_jobs():
+
+    backend = SparkDistributedBackend()
+    max_num_concurrent_tasks = 8
+    backend._get_max_num_concurrent_tasks = MagicMock(return_value=max_num_concurrent_tasks)
+
+    assert backend.effective_n_jobs(n_jobs=None) == 1
+    assert backend.effective_n_jobs(n_jobs=-1) == 8
+    assert backend.effective_n_jobs(n_jobs=4) == 4
+
+    with warnings.catch_warnings(record=True) as w:
```

Comment:
depending on the type of warning, it would be better to use `pytest.mark.filterwarning` instead to decorate the test.

Reply:
Oh sorry I see that you are checking that a warning is raised. You want probably check that the content is what you are expecting